### PR TITLE
Enable reporting when there is no data to load

### PR DESCRIPTION
**Pull request by rwidom**
## Fixes
Fixes https://github.com/WordPress/openverse-catalog/issues/476 by @AetherUnbound 

## Description
There was a test for a specific piece of metadata, but when there is no data to load, the detailed metadata don't apply and aren't generated. So we needed a slightly more expansive check for nothing. In testing, I found another place where a very similar issue comes up, and put in some code for that. 

## Testing Instructions
The original issue has great instructions for replicating. It's a little tricky, because you have to have a situation where there isn't any data to load, and providers are always(ish) adding data. 🥇 Also, I _think_ that if you just put in a future date in the airflow test command, things  get deadlocked. But this morning (4/22/2022) I did `airflow dags test brooklyn_museum_workflow 2022-04-22` and rather than the errors described in the issue, I got:

```
[2022-04-22 09:57:21,053] {base.py:70} INFO - Using connection to: id: slack_notifications. Host: slack, Port: None, Schema: , Login: None, Password: None, extra: {}
[2022-04-22 09:57:21,055] {reporting.py:101} INFO - 
*Provider*: `brooklyn_museum`
*Duration of data pull task*: 2 secs
*Number of records upserted per media type*:
  - `image`: _No data_
[2022-04-22 09:57:21,058] {taskinstance.py:1272} INFO - Marking task as SUCCESS. dag_id=brooklyn_museum_workflow, task_id=report_load_completion, execution_date=20220422T000000, start_date=20220422T094831, end_date=20220422T095721
[2022-04-22 09:57:21,065] {backfill_job.py:386} INFO - [backfill progress] | finished run 1 of 1 | tasks waiting: 0 | succeeded: 4 | running: 0 | failed: 0 | skipped: 2 | deadlocked: 0 | not ready: 0
[2022-04-22 09:57:21,066] {backfill_job.py:851} INFO - Backfill done. Exiting.
```

I had noticed a bunch of failures in `just test` after I made the [changes to reporting.py](https://github.com/WordPress/openverse-catalog/pull/492/commits/a075875e892bde5c3009ba8351a87e138ce4fe06) and some errors in the above provider test as well, which all seemed to be resolved when I made the change to [provider_api_scripts/brooklyn_museum.py](https://github.com/WordPress/openverse-catalog/pull/492/commits/7d952ebb80929aba359c21dbe711d2a67f3aca75). It seems so closely related to this issue, I hope it's ok to include in this PR.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py [hunk 1]
```diff
@@ -40,7 +40,10 @@ def main():
     while condition:
         query_param = _get_query_param(offset=offset)
         objects_batch = _get_object_json(query_param=query_param)
-        logger.debug(len(objects_batch))
+        if objects_batch is None:
+            logger.debug("No data")
+        else:
+            logger.debug(len(objects_batch))
```
Comment (AetherUnbound) [inline openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py:46]: Since we're touching this, perhaps we could make this more descriptive?
```suggestion
        logger.debug(f"Number of batches: {len(objects_batch or [])}")
```

File: openverse_catalog/dags/common/loader/reporting.py [hunk 2]
```diff
@@ -72,12 +72,12 @@ def report_completion(
     # List record count per media type
     media_type_reports = ""
     for media_type, counts in record_counts_by_media_type.items():
-        if not counts.upserted:
+        if counts is None or not counts.upserted:
```
Comment (AetherUnbound) [inline openverse_catalog/dags/common/loader/reporting.py:75]: I think we may also want to add some test cases for when `counts` is `None` - let me know if you'd like some advice or assistance on that!

**Review by AetherUnbound — COMMENTED** [review]
Thank you for the thorough PR description! 🙌🏼 Two comments here, I'm also okay with including the Brooklyn Museum fix in this PR.

For simulating a situation where the counts are `None`, you might be able to get away with editing a provider script just to create the behavior you need. We've done that before with PRs like #479. We don't include the changes in the PR, but they're helpful for other folks who are testing!

File: openverse_catalog/dags/common/loader/reporting.py (hunk 2 above)
Comment (rwidom) [reply openverse_catalog/dags/common/loader/reporting.py:75]: Would you mind taking a look at [test_reporting.py](https://github.com/WordPress/openverse-catalog/pull/492/files#diff-bcf55852f44e45efd0d72a9721e230594272b44290c0923e531705e29e751f97)? I think it was easier than I thought, but let me know if you were expecting something really different!

Comment (AetherUnbound) [reply openverse_catalog/dags/common/loader/reporting.py:75]: That's perfect! I think this PR is ready 😄

**Review by AetherUnbound — APPROVED** [review]
This is fantastic, thanks for contributing this fix!

**Review by stacimc — APPROVED** [review]
🚢 Looks great! Thanks for adding the test as well 😄
